Dashboard/Connectors/Serial/SerialPortWrapper.cs

```diff
@@ -47,16 +62,12 @@ public void Configure(string portName, int baudRate)
             _serialPort.Open();
     }
 
-    public string ReadExisting()
+    private string ReadLine()
     {
         _lastMessageReceived = DateTime.Now;
-        return _serialPort.ReadExisting();
+        return _serialPort.ReadLine();
```

Comment:
This now means we wait for an entire line to be provided before sending it to the parser, avoiding any incomplete lines. This comes with a tradeoff, if done on the main thread this can block causing the UI to not update, so we put it on another thread to avoid this